### PR TITLE
Hearthstone Card implementation

### PR DESCRIPTION
**Pull request by marcosvonstremayr**
Implementación del ejercicio 1 de Hearthstone Card

https://user-images.githubusercontent.com/109096201/179561783-1114b388-5e2c-484e-b448-e1bd42d23a3c.mp4

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: Center(
+          child: ListView(
+              children: [
+                Image.asset(
+                    'assets/images/ysera_card.png',
+                  height: 500,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: both variables can be constants

File: lib/main.dart [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                ),
+                Row(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    const Icon(
+                      Icons.favorite,
+                      color: Colors.pink,
+                      size: 30,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: size can be a constant

File: lib/main.dart [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                    ),
+                    const SizedBox(
+                      width: 10,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: width can be constant

File: lib/main.dart [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+                    ),
+                    Text(
+                      "$_likeCounter",
+                      style: const TextStyle(
+                          color: Colors.black,
+                          fontSize: 30
```
Comment (LeonardoMolleker) [inline lib/main.dart]: fontSize can be constant

File: lib/main.dart [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                ),
+                IntrinsicHeight(
+                  child: Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        Column(
+                          crossAxisAlignment: CrossAxisAlignment.center,
+                          children: const [
+                            Padding(
+                              padding: EdgeInsets.all(8.0),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant here

File: lib/main.dart [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+                              child: Text(
+                                  "Name: Ysera",
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+                                style: TextStyle(
+                                  color: Colors.black,
+                                  fontSize: 24,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+                                ),
+                                textAlign: TextAlign.left,
+                              ),
+                            ),
+                            Padding(
+                              padding: EdgeInsets.all(8.0),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+                              child: Text(
+                                "Type: Minion",
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 10]
```diff
@@ -0,0 +1,193 @@
...
+                                style: TextStyle(
+                                  color: Colors.black,
+                                  fontSize: 24,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 11]
```diff
@@ -0,0 +1,193 @@
...
+                              padding: EdgeInsets.all(8.0),
+                              child: Text(
+                                "Faction: Neutral",
+                                style: TextStyle(
+                                  color: Colors.black,
+                                  fontSize: 24,
+                                ),
+                                textAlign: TextAlign.left,
+                              ),
+                            ),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: this whole widget can be abstracted to no repeat code, also the values can be constants

File: lib/main.dart [hunk 12]
```diff
@@ -0,0 +1,193 @@
...
+                          ],
+                        ),
+                        const VerticalDivider(
+                          color: Colors.black,
+                          thickness: 2,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 13]
```diff
@@ -0,0 +1,193 @@
...
+                        ),
+                        Flexible(
+                          child: Column(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: const [
+                              Padding(
+                                padding: EdgeInsets.all(8.0),
+                                child: Text(
+                                  "Attack: 4",
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 14]
```diff
@@ -0,0 +1,193 @@
...
+                                  style: TextStyle(
+                                    color: Colors.black,
+                                    fontSize: 24,
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 15]
```diff
@@ -0,0 +1,193 @@
...
+                                  ),
+                                  textAlign: TextAlign.left,
+                                ),
+                              ),
+                              Padding(
+                                padding: EdgeInsets.all(8.0),
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 16]
```diff
@@ -0,0 +1,193 @@
...
+                                child: Text(
+                                  "Health: 12",
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart [hunk 17]
```diff
@@ -0,0 +1,193 @@
...
+                                  textAlign: TextAlign.left,
+                                ),
+                              )
+                            ],
+                          ),
+                        )
+                      ]
+                  ),
+                ),
+                const SizedBox(height: 100)
```
Comment (LeonardoMolleker) [inline lib/main.dart]: constant

File: lib/main.dart (hunk 3 above)
Comment (FranciscoGaraventa) [reply lib/main.dart]: Instead to use a SizedBox you can wrap you Text into a Padding, and added some margin top, bottom

File: lib/main.dart (hunk 1 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 3 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

I wrapped the Text Widget into a Padding but added EdgeInsets.only(left: 10) for spacing between Icon and Counter value. No need for margin top or bottom.

File: lib/main.dart (hunk 4 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 6 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 5 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 7 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 9 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 8 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 10 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 11 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 12 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 13 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 14 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 15 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 16 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 17 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+                        Flexible(
+                          child: Column(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: const [
+                              CardText(text: "Atack: 4"),
+                              CardText(text: "Health: 12"),
+                              Padding(
+                                padding: EdgeInsets.all(Constants.cardTextPadding),
+                                child: Text(
+                                    "Ability: At the end of your turn, add a Dream Card to your hand",
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This could be a constant

File: lib/main.dart [hunk 19]
```diff
@@ -0,0 +1,127 @@
...
+                        const VerticalDivider(
+                          color: Constants.verticalDividerColor,
+                          thickness: Constants.verticalDividerThickness,
+                        ),
+                        Flexible(
+                          child: Column(
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: const [
+                              CardText(text: "Atack: 4"),
+                              CardText(text: "Health: 12"),
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This two could also be a constant

File: lib/main.dart [hunk 20]
```diff
@@ -0,0 +1,127 @@
...
+                    mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+                      children: [
+                        Column(
+                          crossAxisAlignment: CrossAxisAlignment.center,
+                          children: const [
+                            CardText(text: "Name: Ysera"),
+                            CardText(text: "Type: Minion"),
+                            CardText(text: "Faction: Neutral",),
+                            CardText(text: "Rarity: Legendary"),
+                            CardText(text: "Cost: 9"),
```
Comment (Juliana-Dimeglio) [inline lib/main.dart]: This texts could be constants

**Review by Juliana-Dimeglio — CHANGES_REQUESTED** [review]

**Review by FranciscoGaraventa — APPROVED** [review]
LG

File: lib/main.dart (hunk 18 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 19 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

File: lib/main.dart (hunk 20 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: Done

**Review by Juliana-Dimeglio — APPROVED** [review]
LGTM

**Pull request merged**